WordPress/Sniffs/WP/UseCapabilitiesNotRolesSniff.php

```diff
@@ -0,0 +1,234 @@
+		'current_user_can'          => 1,
+		'current_user_can_for_blog' => 2,
+		'user_can'                  => 2,
+	);
+
+	/**
+	 * Blacklist of core roles which should not to be used directly.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array Role available in core.
+	 */
+	protected $core_roles = array(
+		'super_admin'   => true,
+		'administrator' => true,
+		'editor'        => true,
+		'author'        => true,
+		'contributor'   => true,
+		'subscriber'    => true,
+	);
+
+	/**
+	 * Whitelist of primitive and meta core capabilities.
+	 *
+	 * To be updated after every major release. Sorted as in capabilities tests.
+	 * Last updated for WordPress 4.9.6.
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/master/tests/phpunit/tests/user/capabilities.php
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array All capabilities available in core.
+	 */
+	protected $core_capabilities = array(
+		'unfiltered_upload'           => true,
+		'unfiltered_html'             => true,
+		'activate_plugins'            => true,
+		'create_users'                => true,
+		'delete_plugins'              => true,
+		'delete_themes'               => true,
+		'delete_users'                => true,
+		'edit_files'                  => true,
+		'edit_plugins'                => true,
+		'edit_themes'                 => true,
+		'edit_users'                  => true,
+		'install_plugins'             => true,
+		'install_themes'              => true,
+		'update_core'                 => true,
+		'update_plugins'              => true,
+		'update_themes'               => true,
+		'edit_theme_options'          => true,
+		'export'                      => true,
+		'import'                      => true,
+		'list_users'                  => true,
+		'manage_options'              => true,
+		'promote_users'               => true,
+		'remove_users'                => true,
+		'switch_themes'               => true,
+		'edit_dashboard'              => true,
+		'moderate_comments'           => true,
+		'manage_categories'           => true,
+		'edit_others_posts'           => true,
+		'edit_pages'                  => true,
+		'edit_others_pages'           => true,
+		'edit_published_pages'        => true,
+		'publish_pages'               => true,
+		'delete_pages'                => true,
+		'delete_others_pages'         => true,
+		'delete_published_pages'      => true,
+		'delete_others_posts'         => true,
+		'delete_private_posts'        => true,
+		'edit_private_posts'          => true,
+		'read_private_posts'          => true,
+		'delete_private_pages'        => true,
+		'edit_private_pages'          => true,
+		'read_private_pages'          => true,
+		'edit_published_posts'        => true,
+		'upload_files'                => true,
+		'publish_posts'               => true,
+		'delete_published_posts'      => true,
+		'edit_posts'                  => true,
+		'delete_posts'                => true,
+		'read'                        => true,
+		'level_10'                    => true,
```

Comment:
If I remember correctly, the use of "level_xx" capabilities was deprecated ages ago. Should these be put in a separate list and warned against when found ?